lib/deep_link_helper.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:firebase_dynamic_links/firebase_dynamic_links.dart';
+
+class DeepLink {
+  final String deepLinkParameter;
+  final Function(String) action;
+
+  DeepLink({
+    this.deepLinkParameter,
+    this.action,
+  });
+}
+
+class DeepLinkHelper {
+  final List<DeepLink> deepLinks;
+
+  DeepLinkHelper({
+    this.deepLinks,
+  });
+
+  void init() {
+    initDynamicLinks();
+  }
+
+  void initDynamicLinks() async {
+    final PendingDynamicLinkData data =
+        await FirebaseDynamicLinks.instance.getInitialLink();
+
+    Uri deepLink = data?.link;
+
+    if (data?.link != null) {
+      _parseDeepLink(deepLink);
+    }
+
+    FirebaseDynamicLinks.instance.onLink(
+      onSuccess: (PendingDynamicLinkData dynamicLink) async {
+        final Uri deepLink = dynamicLink?.link;
+
+        if (deepLink != null) {
+          _parseDeepLink(deepLink);
+        }
+      },
+      onError: (OnLinkErrorException e) async {
+        print('onLinkError');
+        print(e.message);
```

Comment:
We should have these prints in case of error? Or we should have some kind of feedback to the user?

Reply:
We can have a feedback to the user, but in all apps I have seen that if the deep link is wrong simply goes doesn't open it and goes to de home

Reply:
Ok, just to make sure if that was something intended or not.